### DISCUSSION TITLE
Backlinks alphabetical sort

### DISCUSSION THREAD

**Discussion by Elliria** [zim-desktop-wiki/zim-desktop-wiki#2271] [Ideas]
Hey there. I'm using Zim 0.74.3 under Kubuntu 22.04 LTS. The backlinks seem to be sorted by order of creation date, which is fine if I create them in alphabetical order. Where I run into trouble is if I create a new page that's higher up alphabetically than the last page listed in the backlinks pane. It then goes to the bottom of the list so that the list is no longer in alphabetical order.

I tried editing the source of the page and removing the creation date, but the order of the pages listed in the backlinks pane didn't change after doing that, so I put the creation date line back in. Zim must be getting its sorting information from somewhere else. There are a lot of settings in Zim, so it's possible I overlooked the one I need. Is there an option I can tick to sort the backlinks alphabetically by name?

**Comment by introt** [comment]
Hi,

This is the current implementation:

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/48a18f57cb590e02fe05af018f8baf62430fb583/zim/plugins/backlinkpane.py#L95-L120

The backlinks' ordering is that which [list_links](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/48a18f57cb590e02fe05af018f8baf62430fb583/zim/notebook/index/links.py#L227) returns, corresponding to the order of the pages in the database - in other words, they do indeed appear to be sorted by order of creation, or at least of indexing.

To quote Jaap,

> Using the natural_sort() from zim.utils for string sort would solve this elegantly

Feel free to open a feature request.

**Comment by Elliria** [comment]
Thank you for the explanation and suggestion. I opened the issue #2295 as a feature request and linked to this in it.
